### ISSUE TITLE
Emitted 'error' event on Socket instance

### ISSUE THREAD

**Issue by geolffreym** [libp2p/js-libp2p-websockets#114]
```
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: connect ENETUNREACH 2604:1380:1:b600::3:4002 - Local (:::0)
    at internalConnect (net.js:923:16)
    at defaultTriggerAsyncIdScope (internal/async_hooks.js:313:12)
    at net.js:1011:9
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
Emitted 'error' event on Socket instance at:
    at emitErrorNT (internal/streams/destroy.js:84:8)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: -101,
  code: 'ENETUNREACH',
  syscall: 'connect',
  address: '2604:1380:1:b600::3',
  port: 4002
}

```

CC @vasco-santos @jacobheun

**Comment by jacobheun** [comment]
@geolffreym does this occur at startup or while the node is running? Are you able to reliably reproduce this?

`ENETUNREACH` indicates that you don't have network access.

We should still be catching the error though.  It's possible the error listener hasn't been bound yet due to the nature of the issue.

**Comment by geolffreym** [comment]
@jacobheun usually happens when I use websocket as transport and it is happening at runtime...
ipfs@0.47.0 it solved downgrading to 0.46.0

**Comment by jacobheun** [comment]
The websockets library will dial before we can bind an error listener. If the dial fails before that happens this error can be thrown. We need to find a way to properly listen/catch those.
